### DISCUSSION TITLE
is there support planned for dynamic dags

### DISCUSSION THREAD

**Discussion by sudssf** [apache/airflow#13144] [Q&A]
We have usecase where we want to change dag tasks based on some logic at runtime. while doing this backfills & UI element should work properly
for now we are using branching operator and skip tasks which we dont need. but why can we just call task based on run time value or previous task results.

I looked at solution posted on various blogs like using variable, using subdag, using multi_trigger dag etc but none of them seem to handle dynamic dag scenario.

I was hoping that task group will solve this problem but looks like task groups are also created at the time of dag loading and not at the time of creating dag run so dynamic task groups are also not possible.

**Comment by potiuk** [comment]
This is something that we might consider in the future. I know @ashb  had some initial ideas about it but we were all busy to get 2.0.0 out - next few weeks you can expect some concrete ideas and plans for new versions of Airlfow flowing (just watch the devlist).
